sql/src/main/java/org/apache/apex/malhar/sql/planner/RelNodeVisitor.java

```diff
@@ -0,0 +1,457 @@
+      expMap.put(leftName, rightName);
+    }
+    operator.setExpressionMap(expMap);
+    operator.setCondition(expression);
+
+    return new RelInfo("Filter", Lists.<Operator.InputPort>newArrayList(operator.input), operator, operator.output,
+      filter.getRowType());
+  }
+
+  public RelInfo visitSample(Sample sample, List<RelInfo> inputStreams) throws Exception
+  {
+    return defaultValue(sample, inputStreams);
+  }
+
+  public RelInfo visitSort(Sort sort, List<RelInfo> inputStreams) throws Exception
+  {
+    return defaultValue(sort, inputStreams);
+  }
+
+  public RelInfo visitTableModify(TableModify modify, List<RelInfo> inputStreams) throws Exception
+  {
+    Preconditions.checkArgument(modify.isInsert(), "Only INSERT allowed for table modify");
+
+    ApexSQLTable table = modify.getTable().unwrap(ApexSQLTable.class);
+
+    Endpoint endpoint = table.getEndpoint();
+    return endpoint.populateOutputDAG(dag, typeFactory);
+  }
+
+  public RelInfo visitTableScan(TableScan scan, List<RelInfo> inputStreams) throws Exception
+  {
+    ApexSQLTable table = scan.getTable().unwrap(ApexSQLTable.class);
+    Endpoint endpoint = table.getEndpoint();
+    return endpoint.populateInputDAG(dag, typeFactory);
+  }
+
+  public RelInfo visitUncollect(Uncollect uncollect, List<RelInfo> inputStreams) throws Exception
+  {
+    return defaultValue(uncollect, inputStreams);
+  }
+
+  public RelInfo visitWindow(Window window, List<RelInfo> inputStreams) throws Exception
+  {
+    return defaultValue(window, inputStreams);
+  }
+
+  public RelInfo visitJoin(Join join, List<RelInfo> inputStreams) throws Exception
+  {
+    if (inputStreams.size() != 2) {
+      throw new UnsupportedOperationException("Join is a BiRel");
+    }
+
+    if ((join.getJoinType() == JoinRelType.FULL) || (join.getJoinType() == JoinRelType.LEFT) ||
+        (join.getJoinType() == JoinRelType.RIGHT)) {
+      throw new UnsupportedOperationException("Outer joins are not supported");
+    }
+
+    final List<Integer> leftKeys = new ArrayList<>();
+    final List<Integer> rightKeys = new ArrayList<>();
+
+    RexNode remaining =
+        RelOptUtil.splitJoinCondition(join.getLeft(), join.getRight(), join.getCondition(), leftKeys, rightKeys);
+
+    if (leftKeys.size() != rightKeys.size()) {
+      throw new RuntimeException("Unexpected condition reached. Left and right condition count should be same");
+    }
+
+    if (leftKeys.size() == 0) {
+      throw new UnsupportedOperationException("Theta joins are not supported.");
+    }
+
+    RelInfo relInfo = addInnerJoinOperator(join, leftKeys, rightKeys);
+
+    if (!remaining.isAlwaysTrue()) {
+      relInfo = addJoinFilter(join, remaining, relInfo);
+    }
+
+    return relInfo;
+  }
+
+  private RelInfo addJoinFilter(Join join, RexNode remaining, RelInfo relInfo)
+  {
+    FilterTransformOperator operator = dag.addOperator(OperatorUtils.getUniqueOperatorName(join.getRelTypeName() +
```

Comment:
I wanted to but there is a little difference in 2 cases and I wanted to complete out the functionality first. I suggest we take this up later. But good suggestion.